File: corporate-reporting/src/types/ThresholdAppetite.ts
import { IThreshold } from "./Threshold";
import { IEntity, Entity } from "./Entity";

export interface IThresholdAppetite extends IEntity {
    ID: number;
    ThresholdID: number;
    RiskImpactLevelID: number;
    RiskProbabilityID: number;
    Acceptable: boolean;

    Threshold?: IThreshold;
    RiskProbability?: IEntity;
    RiskImpactLevel?: IEntity;

}

export class ThresholdAppetite extends Entity implements IThresholdAppetite {
    public ThresholdID = null;
    public RiskProbabilityID = null;
    public RiskImpactLevelID = null;
    public Acceptable = false;
}

File: ORB.Core/Models/ThresholdAppetite.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class ThresholdAppetite : ObjectWithId
    {
        public int ThresholdID { get; set; }
        public int RiskImpactLevelID { get; set; }
        public int RiskProbabilityID { get; set; }
        public bool Acceptable { get; set; }

        public virtual RiskImpactLevel RiskImpactLevel { get; set; }
        public virtual RiskProbability RiskProbability { get; set; }
        public virtual Threshold Threshold { get; set; }
    }
}
